packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -23,7 +24,119 @@ const restrictedNodeFields = [
 
 exports.setFieldsOnGraphQLNodeType = require(`./extend-node-type`).extendNodeType
 
-exports.onPreBootstrap = validateOptions
```

Comment:
This is what makes this API difficult to add. Removing these pieces means that we won't do any validation in older gatsby versions. So technically, we decrease the user experience for older gatsby users.